### PR TITLE
Fix dead link for Modern Perl book

### LINKED ISSUE #5523: testsuite doc. stale perl book URL
The following link is stale:

https://github.com/i3/i3/blob/a95870120ce30843d95bdc5d87d26973789ed947/docs/testsuite#L40

### PR DESCRIPTION
**Pull request by waterkip**
Closes: #5523

### CODE REVIEW

**Review by orestisfl — COMMENTED** [review]
Given that this book is not available online and seems quite old, maybe it's better to just let test writers select their own Perl resources instead of directing them to potentially outdated info? WDYT @stapelberg

**Comment by stapelberg** [comment]
> Given that this book is not available online and seems quite old, maybe it's better to just let test writers select their own Perl resources instead of directing them to potentially outdated info? WDYT @stapelberg

Hmm, not sure. I think the book is still valuable, though building it from source is not something that people will do, I think. Could we build it from source and distribute a static PDF version of it, or does the license not permit that?

**Comment by orestisfl** [comment]
The book might be useful indeed but instructing contributors on how to learn a language is not really typical for an open source project. Besides, it might be discouraging people rather than helping them; it creates the impression that one has to read a full book before actually contributing to i3 ("Please read the following documents before working on tests"). I believe that for the scope of what we are doing, great knowledge of perl is not needed to contribute to i3. I've personally never read the book.

**Comment by waterkip** [comment]
> Could we build it from source and distribute a static PDF version of it, or does the license not permit that?

You can, the works has a Creative Commons Attribution-Noncommercial-Share
Alike 3.0 United States License.

Although I agree with @orestisfl's statement: The book might be useful indeed but instructing contributors on how to learn a language is not really typical for an open source project.

**Comment by stapelberg** [comment]
> You can, the works has a Creative Commons Attribution-Noncommercial-Share Alike 3.0 United States License.

Okay. I have saved a copy of the PDF, you can refer to it as https://i3wm.org/downloads/modern_perl_a4.pdf

> Although I agree with @orestisfl's statement: The book might be useful indeed but instructing contributors on how to learn a language is not really typical for an open source project.

Yes, but Perl isn’t really a typical language for an open source project either (these days). I’m open to changing the wording to seem less daunting, but I do think it is good to provide a few pointers regarding known-good resources to get into Perl for people who prefer that approach. Of course, everyone is free to disregard the advice and just jump into the code :)

**Comment by waterkip** [comment]
> > You can, the works has a Creative Commons Attribution-Noncommercial-Share Alike 3.0 United States License.
> 
> Okay. I have saved a copy of the PDF, you can refer to it as https://i3wm.org/downloads/modern_perl_a4.pdf

I updated the link in the PR.

> Yes, but Perl isn’t really a typical language for an open source project either (these days). I’m open to changing the wording to seem less daunting, but I do think it is good to provide a few pointers regarding known-good resources to get into Perl for people who prefer that approach. Of course, everyone is free to disregard the advice and just jump into the code :)

Perl is a great language :)

File: docs/testsuite [hunk 1]
```diff
@@ -36,8 +36,8 @@ that, but it will also be useful for every future change.
 
 Apart from this document, you should also have a look at:
 
-1. The "Modern Perl" book, which can be found at
-   http://onyxneon.com/books/modern_perl/modern_perl_a4.pdf
```
Comment (orestisfl) [inline docs/testsuite:40]: This link is pasted in all testcases as well, can you replace it there as well?

You can easily do it with [sd](https://github.com/chmln/sd):

```
sd -s 'http://onyxneon.com/books/modern_perl/modern_perl_a4.pdf' 'https://i3wm.org/downloads/modern_perl_a4.pdf' $(git ls-files) 
```

or sed equivalent

Comment (waterkip) [reply docs/testsuite:40]: Done!

**Review by orestisfl — APPROVED** [review]
Thanks!

**Pull request merged**